lib/remap-functions/src/encode_json.rs

```diff
@@ -0,0 +1,154 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct EncodeJson;
+
+impl Function for EncodeJson {
+    fn identifier(&self) -> &'static str {
+        "encode_json"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |_| true,
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(EncodeJsonFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct EncodeJsonFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for EncodeJsonFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let value = self.value.execute(state, object)?;
+        serde_json::to_string(&value)
+            .map(|v| v.into())
+            .map_err(|e| format!("unable encode to json: {}", e).into())
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .into_fallible(true) // Serialization error
```

Comment:
Just out of curiosity, can you think of any cases where this would actually happen?

Reply:
I do not see how we can get error here, but not sure about this on one hundred percent :confused:

Reply:
:thinking: I'm just wondering as, depending on the error handling strategy we decide on, it might result in us needlessly requiring users to handle errors from `parse_json` even if it would never error in practice. It might be worth digging into serde to see why this would fail. My guess is it would only fail if one of the types wasn't serializable, but that shouldn't be the case for us so we could choose to panic here and save the user from error handling.

Reply:
Checked `serde_json`...  so there will be error if Map key is not a string (but integer is ok too):
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=946f987344f86f66cd36c807557411f2

Both cases will not exists for us, so I think this will be ok to not return error (i.e. panic instead).